live: markdown report support. (#3856)
iterative/dvc.org: content/docs/dvclive/api-reference/live/index.md

- `dir` - Location of the directory to store
  [outputs](/doc/dvclive/get-started#outputs).

- `summary_path` - `{Live.dir}.json`. Location of the

>   [summary](/doc/dvclive/api-reference/live/log#description).

Reviewer: Hola. Why this change? Was just `dir` incorrect? It's listed as just `dir` above.

> Seems unrelated to https://github.com/iterative/dvclive/pull/266 too.

Author: > Seems unrelated to https://github.com/iterative/dvclive/pull/266 too.

It has nothing to do, just something I noticed during this P.R. Could be extracted to other

> - `report_path` - `{Live.dir}/report.{format}`. Location of the
>   [metrics report](/doc/dvclive/api-reference/live/make_report#description).

Reviewer: ```suggestion
- `report_path` - `{Live.dir}/report.{format}`. Location of the
  [metrics report](/doc/dvclive/api-reference/live/make_report). The `{format}`
  can be `html` (HTML) or `md` (Markdown) depending on the value of the
  `report` parameter.
```

  </admon>

- `report` - If `auto`,`html` or `md`, DVCLive will call `Live.make_report()` on

>   each step update. _Default_: `auto`.

Reviewer: Please explain what `auto` does.

Reviewer: Nvmd I see it's down here...

> - `report` - If `auto`,`html` or `md`, DVCLive will call `Live.make_report()` on
>   each step update. _Default_: `auto`.

Reviewer: ```suggestion
- `report` - If `auto`,`html`, or `md`, DVCLive will call `Live.make_report()` on
  each step update. The `auto` mode (default) will use `md` format if a `CI` env
  var is present, otherwise it will use `html`.
```

- `report` - If `auto`,`html` or `md`, DVCLive will call `Live.make_report()` on
  each step update. _Default_: `auto`.

- `auto_open` - If `True`, on the first `Live.make_report()` call, DVCLive will

>   automatically open `html_path` in a browser. _Default_: `False`.

Reviewer: Wait what happened to `auto_open`?

Author: Docs were outdated. I just noticed this was still here during this P.R. 
It was removed a few versions ago, discussion here https://github.com/iterative/dvclive/issues/241

>   The `auto` mode will use `md` format if `CI` env var is present, otherwise, it
>   will use `html`.
> 
>   If `report` is `None`, `Live.make_report()` won't be called on each step
>   update.

Reviewer: ```suggestion
  If `report` is `None`, `Live.make_report()` won't be called automatically.
```
